test/test_repo.py

```diff
@@ -263,6 +266,40 @@ def test_leaking_password_in_clone_logs(self, rw_dir):
             to_path=rw_dir,
         )
 
+    def test_unsafe_options(self):
+        self.assertFalse(Repo.unsafe_options("github.com/deploy/deploy"))
+
+    def test_unsafe_options_ext_url(self):
+        self.assertTrue(Repo.unsafe_options("ext::ssh"))
+
+    def test_unsafe_options_multi_options_upload_pack(self):
+        self.assertTrue(Repo.unsafe_options("", ["--upload-pack='touch foo'"]))
```

Comment:
I have trouble understanding this as I don't see `unsafe_options()` in the diff.

Assuming it's all good and right though, can it also detect ` self.assertTrue(Repo.unsafe_options("", ["--upload-pack" "touch foo"]))` ?

Reply:
These test are from the other PR, I need to update/change them